### DISCUSSION TITLE
Seat part

### DISCUSSION THREAD

**Discussion by TW1STaL1CKY** [CapsAdmin/pac3#989] [Ideas]
Part that enables sitting much like how VRChat avatars can have seats on.
Example: a horse PAC outfit with a seat part on the saddle - another player can press use on the seat on the saddle and they will sit on the saddle.

It would need to be serverside to an extent to create the seat and parent it to a certain position so using it works. I don't think it would be exploitable, at least no more than projectiles can be exploited.

Possible additional tasks:
- Use _RenderOverride_ or _PrePlayerDraw_ to accurately update where the sitting player sits
- Disable or restrict scaling on the seat so players can't make their hitboxes (use range) bigger than normal
- Allow adding the bone parts (any parts?) to the player sitting on the seat part ([idea by @Earu](https://github.com/CapsAdmin/pac3/issues/977#issuecomment-770430484))
    - If this is done, maybe add a convar that lets players sitting on seat parts not allow seat parts to apply changes to them?

**Comment by Earu** [comment]
I approve of this, I'll repost what I wrote on Discord:

> hey @CapsAdmin heres an idea from VRChat and some other friends for a pac part, a vehicle part that allows you to create seats on your outfit, and let people use them
> Also have a specific group tied to the seat where you can set the bones and whatnot of the player sitting

**Comment by GenericcNameHere** [comment]
I realise this discussion post is fairly old, but I was thinking of this same idea recently and would love to see it.
Another feature that could be added on to it could being able to select what seat animation plays for the sitting player, for example the regular sitting animation, the Jeep and Airboat animations and even using the Pod animation, perhaps even being able to link custom animations to it?

It would be a hell of a lot better than just using that sit anywhere addon where sitting on a player just sits on top of their collision hull.

**Comment by thegrb93** [comment]
I think the problem sit-anywhere faces, as well as this addition, is that it's not possible to parent entities to specific bones in the serverside. Attachments I think are allowed, but not bones for whatever reason. It's annoyed me countless times in the past.
